gsoc_blog_setup.md

```diff
@@ -0,0 +1,131 @@
+Once you have been accepted as a GSoC student NumFOCUS expects you to blog about your progress every week. It's okay if you are stuck, post about that too :) You could use various platforms like wordpress, blogger, etc. but in our experience it's usually a pain to get formatting right in case of code snippets, screenshots. We suggest you use GitHub Pages and Jekyll to blog about your summer. 
+GitHub gives you free hosting and a free domain name (`github_username.github.io`) and [Jekyll](https://jekyllrb.com) is a blog friendly static site generator. This tutorial will help you setup your blog.
+
+#### Install Ruby
+For Debian or Ubuntu based users
+```
+$ sudo apt-get install ruby-full
+```
+Mac OS/OS X has ruby installed by default.
+
+For more information go through https://www.ruby-lang.org/en/documentation/installation/
+
+#### Install RVM and setup 
+```
+$ \curl -sSL https://get.rvm.io | bash -s stable
+ ```
+for more information https://rvm.io
+
+```
+$ rvm install ruby  #installs the latest version
+$ rvm gemset create blog     #creates a new gemset (ruby environment)
+$ rvm gemset use blog       #change to blog gemset
+$ gem install bundler jekyll
+```
+#### Create the blog site
+
+```
+$ jekyll new gsoc_blog
+$ cd gsoc_blog
+$ bundle exec jekyll serve
+```
+
+You should have a local server running on port 4000 (http://127.0.0.1:4000/) with a sample post on the blog. This uses the default theme, there are various others themes available feel free to hack around.
+
+#### Integrate with GitHub Pages
+
+Initialize a git repository inside the gsoc_blog folder.
+```
+$ git init .      # make sure you are inside the gsoc_blog folder
+```
+`git status` should give you the output something like
+```
+On branch master
+
+Initial commit
+
+Untracked files:
+  (use "git add <file>..." to include in what will be committed)
+
+	.gitignore
+	Gemfile
+	Gemfile.lock
+	_config.yml
+	_posts/
+	about.md
+	index.md
+
+nothing added to commit but untracked files present (use "git add" to track)
+```
+Make this as your first commit
+```
+$ git add .
+$ git commit -m 'Initial commit'
+```
+
+Now go to GitHub and create a new repository and name it `github_username.github.io`. **Don't** add readme or license files. If you are already using GitHub pages for a personal website create a new repository and name it `gsoc_blog`. Make sure to turn on GitHub pages in settings for the repository. You can access the blog at `github_username.github.io/gsoc_blog`
```

Comment:
This is in case if you already have a GitHub page on `username.github.io` and want to create a blog, for example I already have a webpage https://mriduls.github.io and if I want to create a new blog I need to create a new repo `blog/gsoc_blog` and the GitHub page url will be something https://mriduls.github.io/blog/